examples/AffinePlaneCurve.jl

```diff
@@ -0,0 +1,148 @@
+module AffinePlaneCurveModule
+
+using Oscar, Markdown
+
+export AffinePlaneCurve, degree, jacobi_ideal, tangent, issmooth_point, union,
+       Point
+
+################################################################################
+#
+# We follow the lecture notes by Andreas Gathmann: 
+# https://www.mathematik.uni-kl.de/~gathmann/en/curves.php
+#
+################################################################################
+mutable struct AffinePlaneCurve{S}
+  eq::Oscar.MPolyElem{S}                  # Equation of the curve (polynomial in two variables)
+  degree::Int64                           # degree of the equation of the curve
+  dimension::Int64                        # dimension of the curve (as a variety)
+  function AffinePlaneCurve(eq::Oscar.MPolyElem{S}) where {S <: FieldElem}
+    r = new{S}()
+    r.eq = eq
+    if nvars(parent(r.eq)) != 2           
+       error("The defining equation must belong to a ring with two variables")
+    else
+       r.degree = -1                      # -1 when it is not computed yet
+       r.dimension = 1                    # since C is a plane curve, the dimension is always 1
+       return r
+    end
+  end
+end
+
+function Base.show(io::IO, C::AffinePlaneCurve)
+  println(io, "Affine plane curve defined by ", C.eq)
+end
+
+################################################################################
+# Point (first attempt).
+# The point is described by its coordinates.
+# ambient_dim gives the dimension of the space to which belongs the point.
+
+mutable struct Point{S}
+  coord::Array{S, 1}
+  ambient_dim::Int64
+  function Point(coordinates::Array{S,1}) where {S <: FieldElem}
```

Comment:
You may also consider spaces after commas. You have them in some places and not others.